File: factorial.java
/**
 * factorial
 */
public class factorial {
    static int fact(int n) {
        if (n == 0 || n == 1) {
            return 1;
        } else {
            return n * fact(n - 1);
        }
    }

    public static void main(String[] args) {
        int n = 7;
        System.out.println(fact(n));
    }
}

File: Csharp/factorial.cs
using System;

namespace factorial
{
    class Program
    {
        public static int Factorial(int num)
        {
            if (num == 0)
                return 1;
            else
                // Recursive call to calculate factorial
                return num * Factorial(num - 1);
        }

        public static void Main(string[] args)
        {
            Console.Write("Enter a number: ");
            int num = int.Parse(Console.ReadLine());

            int fact = Factorial(num);
            Console.WriteLine("Factorial of " + num + " is " + fact);
            
                                    Console.ReadLine();

        }
    }
}
